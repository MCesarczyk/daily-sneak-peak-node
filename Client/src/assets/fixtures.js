export const groups = [
  { id: 1, label: "Cangaroos" },
  { id: 2, label: "Froggies" },
  { id: 3, label: "Crocodiles" },
  { id: 4, label: "Slothies" },
  { id: 5, label: "Monkeys" }
];