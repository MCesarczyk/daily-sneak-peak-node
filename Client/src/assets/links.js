export const TITLE_URL = "/api/title";
export const CHILDREN_LIST_URL = "/api/get/children";
export const CHILD_POST_URL = "/api/post/childtodb";
export const CHILD_UPDATE_URL = "/api/put/children";