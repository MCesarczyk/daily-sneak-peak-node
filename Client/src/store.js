import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "@redux-saga/core";
import childrenReducer from "./views/children/childrenSlice";
import rootSaga from "./rootSaga";

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: {
    children: childrenReducer,
  },
  middleware: [sagaMiddleware],
});

sagaMiddleware.run(rootSaga);

export default store;