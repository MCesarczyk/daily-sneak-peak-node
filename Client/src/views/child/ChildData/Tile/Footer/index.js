import React from "react";
import { Divider } from "@mui/material";
import { ButtonsWrapper, StyledFooter } from "./styled";
import { useDispatch, useSelector } from "react-redux";
import { deleteChildData, selectChildData } from "../../../childSlice";

const TileFooter = () => {
  const dispatch = useDispatch();
  const child = useSelector(selectChildData);

  return (
    <StyledFooter>
      <Divider />
      <ButtonsWrapper>
      </ButtonsWrapper>
    </StyledFooter>
  )
};

export default TileFooter;