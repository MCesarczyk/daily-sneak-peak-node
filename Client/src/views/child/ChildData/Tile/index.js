import React from "react";
import { Skeleton } from "@mui/material";
import TileFooter from "./Footer";
import NotFound from "../../../../components/NotFound";
import { StyledTile } from "./styled";
import Image from "./Image";
import Name from "./Name";

const Tile = ({ child }) => (
  <>
    {!child
      ? <NotFound message="Sorry... no child data found." />
      : <StyledTile data-testid="child-tile">
        <Image />
        {child && Object.entries(child).length === 0 ?
          <>
            <Skeleton height={42} />
            <Skeleton height={36} />
          </>
          :
          <Name
            name={child?.name + ' ' + child?.surname}
            group={child?.group}
          />
        }
        <TileFooter />
      </StyledTile>
    }
  </>
);

export default Tile;