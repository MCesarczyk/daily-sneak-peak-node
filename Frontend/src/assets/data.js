export const AUTHOR_PAGE_ADDRESS = "https://home.cesarczyk.dev";
export const AUTHOR_PAGE_LABEL = "Cesarczyk.dev";
export const DEMO_DELAY = 300;