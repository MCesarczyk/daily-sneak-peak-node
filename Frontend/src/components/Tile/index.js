import React from "react";
import { Skeleton } from "@mui/material";
import TileFooter from "./Footer";
import NotFound from "../NotFound";
import { StyledTile } from "./styled";
import Avatar from "./Avatar";
import Heading from "./Heading";
import Loader from "../Loader";

const Tile = ({ state, componentsState, avatarUrl, headingData, footerData }) => (
  <Loader
    state={state}
    message="Loading"
  >
    {componentsState.notFound
      ? <NotFound message={componentsState.errorMessage} />
      : <StyledTile data-testid="details-tile">
        <Avatar url={avatarUrl} />
        {componentsState.loading ?
          <>
            <Skeleton height={42} />
            <Skeleton height={36} />
          </>
          :
          <Heading data={headingData} />
        }
        <TileFooter data={footerData} />
      </StyledTile>
    }
  </Loader>
);

export default Tile;