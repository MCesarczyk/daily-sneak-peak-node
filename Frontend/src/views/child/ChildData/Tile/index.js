import React from "react";
import { Skeleton } from "@mui/material";
import TileFooter from "./Footer";
import NotFound from "../../../../components/NotFound";
import { StyledTile } from "./styled";
import Image from "./Image";
import Heading from "./Heading";

const Tile = ({ child, headingData, footerData }) => (
  <>
    {!child
      ? <NotFound message="Sorry... no child data found." />
      : <StyledTile data-testid="child-tile">
        <Image child={child} />
        {child && Object.entries(child).length === 0 ?
          <>
            <Skeleton height={42} />
            <Skeleton height={36} />
          </>
          :
          <Heading data={headingData} />
        }
        <TileFooter data={footerData} />
      </StyledTile>
    }
  </>
);

export default Tile;