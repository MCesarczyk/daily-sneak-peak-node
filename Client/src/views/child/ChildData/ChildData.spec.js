import reducer from "../childSlice";
import mock from "../../../assets/mocks/child.json";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../store";
import { ThemeProvider } from "styled-components";
import { lightMode } from "../../../assets/theme";
import Tile from "./Tile";

const setup = (child) => render(
  <Provider store={store}>
    <ThemeProvider theme={lightMode}>
      <Tile child={child} />
    </ThemeProvider>
  </Provider>
);

test("Child reducer should return the initial state", () => {
  expect(reducer(undefined, {})).toEqual(
    {
      id: 0,
      data: {},
      gotoList: false
    }
  )
});

test("Child reducer should return mocked data", () => {
  const previousState = mock;
  expect(reducer(previousState, {})).toMatchSnapshot();
});

test("Child data should render properly", async () => {
  setup(mock);

  const tile = screen.queryByTestId('child-tile');
  expect(tile).toBeInTheDocument();

  expect(screen.getByText(/name:/)).toBeInTheDocument();
  expect(screen.getByRole('heading').innerHTML).toBe('Mary Jane');
  expect(screen.getByText(/group:/)).toBeInTheDocument();
  expect(screen.getByRole('subheading').innerHTML).toBe("Crocodiles");
});

test("Placeholder should display when it's no data", () => {
  setup();

  const tile = screen.queryByTestId('child-tile');
  expect(tile).not.toBeInTheDocument();

  const errorMessage = screen.getByRole('heading');
  expect(errorMessage.innerHTML).toBe('Sorry... no child data found.');

  const errorImage = screen.queryByRole('img');
  expect(errorImage).toBeInTheDocument();
});