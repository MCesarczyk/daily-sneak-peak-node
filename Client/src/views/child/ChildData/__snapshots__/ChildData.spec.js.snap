// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Child reducer should return mocked data 1`] = `
Object {
  "created_at": "2022-01-24T00:08:31.474Z",
  "group": "Crocodiles",
  "id": 2,
  "name": "Mary",
  "surname": "Jane",
  "updated_at": "2022-01-24T00:08:31.474Z",
}
`;
